Scripts/Runtime/Util/Systems/PlayerLoopUtil.cs

```diff
@@ -0,0 +1,117 @@
+        public static ComponentSystemGroup GetSystemGroupFromPlayerLoopSystem(ref PlayerLoopSystem playerLoopSystem)
+        {
+            if (playerLoopSystem.updateDelegate?.Target == null
+             || !COMPONENT_SYSTEM_GROUP_TYPE.IsAssignableFrom(playerLoopSystem.type))
+            {
+                return null;
+            }
+
+            return GetSystemGroupFromPlayerLoopSystemNoChecks(ref playerLoopSystem);
+        }
+        
+        /// <summary>
+        /// Given a <see cref="PlayerLoopSystem"/>, gets the <see cref="ComponentSystemGroup"/> associated.
+        /// This method has no checks and assumes that you are certain the <see cref="PlayerLoopSystem"/>
+        /// does contain a <see cref="ComponentSystemGroup"/>. 
+        /// </summary>
+        /// <param name="playerLoopSystem">The <see cref="PlayerLoopSystem"/> to use.</param>
+        /// <returns>
+        /// The <see cref="ComponentSystemGroup"/> associated.
+        /// </returns>
+        public static ComponentSystemGroup GetSystemGroupFromPlayerLoopSystemNoChecks(ref PlayerLoopSystem playerLoopSystem)
+        {
+            return (ComponentSystemGroup)GetSystemFromPlayerLoopSystemNoChecks(ref playerLoopSystem);
+        }
+
+        /// <summary>
+        /// Given a <see cref="PlayerLoopSystem"/>, gets the <see cref="ComponentSystemBase"/> associated.
+        /// This is a "safe" method, it does not assume there is a <see cref="ComponentSystemBase"/> to get
+        /// and may return null.
+        /// </summary>
+        /// <param name="playerLoopSystem">The <see cref="PlayerLoopSystem"/> to use.</param>
+        /// <returns>
+        /// The <see cref="ComponentSystemBase"/> if it exists.
+        /// null if this <see cref="PlayerLoopSystem"/> is a phase
+        /// null if this <see cref="PlayerLoopSystem"/> does not have a <see cref="ComponentSystemBase"/>
+        /// </returns>
+        public static ComponentSystemBase GetSystemFromPlayerLoopSystem(ref PlayerLoopSystem playerLoopSystem)
+        {
+            if (playerLoopSystem.updateDelegate?.Target == null)
+            {
+                return null;
+            }
+
+            return GetSystemFromPlayerLoopSystemNoChecks(ref playerLoopSystem);
+        }
+
+        /// <summary>
+        /// Given a <see cref="PlayerLoopSystem"/>, gets the <see cref="ComponentSystemBase"/> associated.
+        /// This method has no checks and assumes that you are certain the <see cref="PlayerLoopSystem"/>
+        /// does contain a <see cref="ComponentSystemBase"/>. 
+        /// </summary>
+        /// <param name="playerLoopSystem">The <see cref="PlayerLoopSystem"/> to use.</param>
+        /// <returns>
+        /// The <see cref="ComponentSystemBase"/> associated.
+        /// </returns>
+        public static ComponentSystemBase GetSystemFromPlayerLoopSystemNoChecks(ref PlayerLoopSystem playerLoopSystem)
+        {
+            object wrapper = playerLoopSystem.updateDelegate.Target;
```

Comment:
You're checking that this value is null in all the other calls (Ex: GetSystemFromPlayerLoopSystem) you probably should here as well since it'll throw an exception if it is.

Although I see it's a "no checks" variant...hmmm. Adding a check would reduce the number of checks from other other methods in this class.

Reply:
This is an optimized path for a caller to take on the risk for that exception. 

If you don't know, you call the safe version `TryGetSystemFromPlayerLoopSystem`

If you do know, for example like in `WorldCache.ParsePlayerLoopSystem` where we've already done that check, you can call the optimized version `TryGetSystemFromPlayerLoopSystemNoChecks` 

This is similar to unity's naming in things like `NativeArrayUnsafeUtility.GetUnsafeBufferPointerWithoutChecks`